src/Common/Commands.Common/CustomAttributes/BreakingChangeAttributeHelper.cs

```diff
@@ -0,0 +1,151 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+#undef DEBUG
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.WindowsAzure.Commands.Common.CustomAttributes
+{
+    public class BreakingChangeAttributeHelper
+    {
+        public const String SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME = "SuppressAzurePowerShellBreakingChangeWarnings";
+
+        public const String BREAKING_CHANGE_DOCUMENTATION_HEADER = @"<!--
+    Please leave this section at the top of the breaking change documentation.
+
+    New breaking changes should go under the section titled 'Current Breaking Changes', and should adhere to the following format:
+    ## Current Breaking Changes
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+    # Old
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    ## Release X.0.0
+
+    The following cmdlets were affected this release:
+
+    **Cmdlet 1**
+    - Description of what has changed
+
+    ```powershell
+
+    # Old\
+    # Sample of how the cmdlet was previously called
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    Note: the above sections follow the template found in the link below:
+
+    https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
+    -->";
+
+        public static void processCustomAttributesAtRuntime(Type type, List<String> boundParameterNames)
+        {
+            bool emitWarningOrError = true;
+
+            try
+            {
+                emitWarningOrError = Boolean.Parse(System.Environment.GetEnvironmentVariable(SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
+
+            if (!emitWarningOrError)
+            {
+                //Do not process the attributes at runtime... The env variable to override the warning messages is set
+                return;
+            }
+
+            List<GenericBreakingChangeAttribute> attributes = getAllBreakingChangeAttributesInType(type, boundParameterNames);
+
+            if (attributes != null && attributes.Count > 0)
+            {
+                Console.WriteLine("Breaking changes in the cmdlet : " + Utilities.getNameFromCmdletType(type) + "\n");
```

Comment:
We should not write to the console - is this meant to be tracing info?  If so, we should write to the 'Debug' stream (```cmdlet.WriteDebug()```)